src/Core/Configurations/RuntimeConfigValidatorUtil.cs

```diff
@@ -0,0 +1,65 @@
+using System.Text.RegularExpressions;
+
+namespace Azure.DataApiBuilder.Core.Configurations;
+
+/// <summary>
+/// This class encapsulates methods to validate the runtime config file.
+/// </summary>
+public static class RuntimeConfigValidatorUtil
+{
+    // Reserved characters as defined in RFC3986 are not allowed to be present in the
+    // REST/GraphQL custom path because they are not acceptable to be present in URIs.
+    // Refer here: https://www.rfc-editor.org/rfc/rfc3986#page-12.
+    private static readonly string _reservedUriChars = @"[\.:\?#/\[\]@!$&'()\*\+,;=]+";
+
+    //  Regex to validate rest/graphql custom path prefix.
+    public static readonly Regex _reservedUriCharsRgx = new(_reservedUriChars, RegexOptions.Compiled);
+
+    public const string URI_COMPONENT_WITH_RESERVED_CHARS_ERR_MSG = "contains one or more reserved characters.";
+
+    /// <summary>
+    /// Method to validate that the REST/GraphQL URI component is well formed and does not contain
+    /// any reserved characters. In case the URI component is not well formed the exception message containing
+    /// the reason for ill-formed URI component is returned. Else we return an empty string.
+    /// </summary>
+    /// <param name="uriComponent">path prefix/base route for rest/graphql apis</param>
+    /// <returns>false when the URI component is not well formed.</returns>
+    public static bool TryValidateUriComponent(string? uriComponent, out string exceptionMessageSuffix)
+    {
+        exceptionMessageSuffix = string.Empty;
+        if (string.IsNullOrEmpty(uriComponent))
+        {
+            exceptionMessageSuffix = "cannot be null or empty.";
+        }
+        // A valid URI component should start with a forward slash '/'.
+        else if (!uriComponent.StartsWith("/"))
+        {
+            exceptionMessageSuffix = "should start with a '/'.";
+        }
+        else if (uriComponent.Any(x => Char.IsWhiteSpace(x)))
+        {
+            exceptionMessageSuffix = "updated Path contains white spaces.";
```

Comment:
update the exception message so that it follows the existing pattern. This method is a generic method for validating Urls, it shouldn't give messages such as `updated path`.

Reply:
updated